wom/src/main/scala/wdl/Scatter.scala

```diff
@@ -36,22 +35,27 @@ object Scatter {
     new Scatter(index, item, WdlExpression(ast.getAttribute("collection")), ast)
   }
 
-  def womScatterNode(scatter: Scatter, localLookup: Map[String, GraphNodePort.OutputPort]): ErrorOr[ScatterNodeWithInputs] = {
+  def womScatterNode(scatter: Scatter, localLookup: Map[String, GraphNodePort.OutputPort]): ErrorOr[ScatterNodeWithNewNodes] = {
+    // Convert the scatter collection WdlExpression to a WdlWomExpression 
     val scatterCollectionExpression = WdlWomExpression(scatter.collection, Option(scatter))
-    val scatterVariableSourceValidation = WdlWomExpression.findInputsforExpression(scatter.item, scatterCollectionExpression, localLookup, Map.empty)
+    // Generate an ExpressionNode from the WdlWomExpression
+    val scatterVariableSourceValidation = WdlWomExpression.toExpressionNode(scatter.item, scatterCollectionExpression, localLookup, Map.empty)
```

Comment:
This variable name is a little strange; it is validating, but it's also producing the scatter collection expression.